awx/main/scheduler/dag_workflow.py

```diff
@@ -213,12 +210,32 @@ def mark_dnr_nodes(self):
 
         for node in self.sort_nodes_topological():
             obj = node['node_object']
-
-            if obj.do_not_run is False and not obj.job and node not in root_nodes:
-                parent_nodes = [p['node_object'] for p in self.get_dependents(obj)]
+            parent_nodes = [p['node_object'] for p in self.get_parents(obj)]
+            if obj.do_not_run is False and not obj.job and node not in root_nodes and not obj.all_parents_must_converge:
```

Comment:
I kind of feel like this conditional and the one below can be consolidated usefully, since they both have the `obj.do_not_run` and `obj.job` clauses in common.  Hard to say, though, there's a lot going on here.

Reply:
updated to be more concise